src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -248,6 +253,8 @@ private void handleResponse() {
             }
         }
 
+        proxy.getRateLimiter().unregister(context);
```

Comment:
is it still necessary? you always unregister in response method

Reply:
I moved unregister call to the future callback of proxy response